#include "mainwidget.h"
#include "ui_mainwidget.h"
#include <QDebug>

MainWidget::MainWidget(QWidget *parent) :
    QWidget(parent),
    ui(new Ui::MainWidget)
{
    ui->setupUi(this);

    socket = new TcpSocket(this);

    connect(socket, &TcpSocket::dealMessage,
            this, &MainWidget::displayMessage);

    timer = new QTimer(this);
    connect(timer, &QTimer::timeout,
            this, &MainWidget::sendEcgWava);

    ui->ipLineEdit->setText("127.0.0.1");

}

MainWidget::~MainWidget()
{
    delete ui;
}

// 发起连接
void MainWidget::on_connectButton_clicked()
{
    QString strIp = ui->ipLineEdit->text();
    quint16 port = ui->portLineEdit->text().toUInt();
    int ret = socket->connectToServer(strIp, port);
    if (ret)
    {
       qDebug() << "连接成功";
       ui->connectButton->setEnabled(false);

       timer->start(1000);
    }
    else
    {
        qDebug() << "连接失败";
    }
}

static const unsigned short int Ecg2_500DemoData[] = {
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2008, 2016, 2016, 2016, 2024, 2032, 2048,
2064, 2064, 2064, 2072, 2080, 2080, 2080, 2088, 2096, 2104,
2112, 2112, 2112, 2112, 2112, 2112, 2104, 2096, 2088,
2080, 2080, 2080, 2072, 2064, 2064, 2064, 2048, 2032, 2032,
2032, 2016, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 1992, 1984, 1976,
1968, 1960, 1952, 1944, 1936, 1944, 1952, 2016, 2080, 2136,
2192, 2256, 2320, 2376, 2432, 2488, 2544, 2568, 2592, 2536,
2480, 2424, 2368, 2304, 2240, 2184, 2128, 2072, 2016, 1968,
1920, 1928, 1936, 1944, 1952, 1960, 1968, 1984, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2008, 2016, 2024, 2032, 2032,
2032, 2048, 2064, 2064, 2064, 2072, 2080, 2088, 2096, 2104,
2112, 2112, 2112, 2120, 2128, 2136, 2144, 2152, 2160, 2160,
2160, 2160, 2160, 2168, 2176, 2176, 2176, 2184, 2192,
2192, 2192, 2192, 2200, 2208, 2208, 2208, 2208, 2208, 2208,
2208, 2200, 2192, 2192, 2192, 2184, 2176, 2176, 2176, 2168,
2160, 2160, 2160, 2144, 2128, 2128, 2128, 2128, 2128, 2112,
2096, 2088, 2080, 2072, 2064, 2064, 2064, 2048, 2032, 2024,
2016, 2016, 2016, 2008, 2000, 2000, 2000, 2000, 2000,
2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000
};

void MainWidget::sendEcgWava()
{
    char* buffer = new char[1024];
    char serial[24] = "dev-01";

    // 内存处理字节,大小端（忽略）
    memcpy(buffer, serial, 24);
    memcpy(buffer+24, Ecg2_500DemoData, 1000);


    socket->sendMessage(buffer);

}



void MainWidget::displayMessage(QString content)
{
    ui->recvLineEdit->setText(content);

}
